src/app/[entity]/[repo]/issues/new/page.tsx

```diff
@@ -1,7 +1,284 @@
+            <span>
+              <span className='text-grey-300/20'>No one yet ––</span>
+              <span className='cursor-pointer text-purple-400'> assign yourself</span>
+            </span>
+          </div>
+        </div>
+        <div className='flex'>
+          <div className='flex flex-col w-full p-2'>
+            <div className='flex hover:text-purple-400 cursor-pointer'>
+              <p className='w-full mb-2'>Labels</p>
+              <div className="hidden items-center md:inline">
+                <DropdownMenu>
+                  <DropdownMenuTrigger asChild>
+                    <div className="flex items-center cursor-pointer">
+                      <Settings />
+                      <ChevronDown className="mt-1 h-4 w-4 hover:text-white/80" />
+                    </div>
+                  </DropdownMenuTrigger>
+                  <DropdownMenuContent className="w-56">
+                    <DropdownMenuLabel>Apply labels to this issue.</DropdownMenuLabel>
+                    <DropdownMenuSeparator />
+                    <Input
+                      id="labelsFilter"
+                      name="labelsFilter"
+                      type="text"
+                      placeholder="Filter labels"
+                      className="w-full block"
+                      ref={labelsFilterRef}
+                    />
+                    <DropdownMenuSeparator />
+                    {/* todo: fetch labels of this repo and replace mockLabels */}
+                    {/* todo: publish labels with NostrContext when the dropdown menu hides */}
+                    {/* todo: filter labels based on labelsFilterRef */}
+                    <DropdownMenuGroup>
+                      {mockLabels.map(label => {
+                        return (
+                          <div key={label}>
+                            <div className='flex cursor-pointer p-1' onClick={() => selectLabel(label)}>
+                              <span className='mr-1'>{label}</span>
+                              {selectedLabels.includes(label) && <Check />}
+                            </div>
+                            <DropdownMenuSeparator />
+                          </div>
+                        )
+                      })}
+                      <div className='flex cursor-pointer p-1'>
+                        <Edit />
+                        { /* todo: link to the /labels page */}
+                        <Link href={"#"} className="ml-2">{'Edit labels'}</Link>
+                      </div>
+                    </DropdownMenuGroup>
+                  </DropdownMenuContent>
+                </DropdownMenu>
+              </div>
+            </div>
+            <div className='flex gap-2'>
+              {
+                selectedLabels.length > 0 ? (
+                  /* todo: different bg-color for each label */
+                  selectedLabels.map(label => <span className='p-1 pl-2 pr-2 text-black bg-white sm:rounded-lg'>{label}</span>)

```

Comment:
needs a key prop here.  I used to use index, but then found out using index was a bad pratice in cases where the order could change.  using the label as the key isn't great because of the possiblity the label is repeated.

in this case of an array of strings, I'm not sure the best way for key.  suggestions @pierresisson ???

Reply:
I think duplicate labels will not be an issue because generally duplicate labels are removed quite quickly.